docs/diagnostics/PX1078.md

```diff
@@ -0,0 +1,51 @@
+# PX1078
+
+This document describes the PX1078 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                   | Type  | Code Fix    |
+| ------ | --------------------------------------------------- | ----- | ----------- |
+| PX1078 | DAC field and referenced field have different size  | Error | Unavailable |
+| PX1078 | DAC field and referenced field have different types | Error | Unavailable |
+
+## Diagnostic Description
+
+The diagnostic verifies the following rules for properties referencing foreign field:
+
+- the property should have same CLR type as foreign property,
+- the property and foreign property should have same size, if data type attribute allows specifying one.
+
+The diagnostic doesn't check following edge cases:
+
+- either of properties have several data type attributes with different lengths,
+- either of properties have data type attributes incompatible with CLR type,
+- foreign property doesn't have data type attribute specified.
+
+## Example of Incorrect Code
+
+```C#
+public class DAC: PXBqlTable, IBqlTable
+{
+    #region Property
+    public abstract class property : IBqlField { }
+ 
+	[PXDBString(16)] // The PX1078 error is displayed for this line.
+	[PXSelector(typeof(SYSubstitution.substitutionID))] // The property have data type attribute `[PXDBString(25)]`
+    public virtual string Property { get; set; }
+    #endregion
+
+    #region OtherProperty
+    public abstract class otherProperty : IBqlField { }
+ 
+	[PXDBInt] // The PX1078 error is displayed for this line.
+	[PXSelector(typeof(SYSubstitution.substitutionID))] // The property have data type attribute `[PXDBString(25)]`
+    public virtual int? OtherProperty { get; set; }
+    #endregion
+}
+```
+
+## Related Articles
+
+ - [Bound Field Data Types](https://help.acumatica.com/Help?ScreenId=ShowWiki&pageid=61059393-8873-451f-b474-783906330fc6)
```

Comment:
I'm not sure about the related articles. Imho, it would be better to add article about `PXDBString` here. But I will leave it up to the doc team.